apps/api/src/persistence/repositories/arango-command-repository-for-aggregate.ts

```diff
@@ -0,0 +1,124 @@
+import { AGGREGATE_COMPOSITE_IDENTIFIER, ICommandBase } from '@coscrad/api-interfaces';
+import { Inject } from '@nestjs/common';
+import { BaseEvent } from '../../domain/models/shared/events/base-event.entity';
+import { Song } from '../../domain/models/song/song.entity';
+import { IRepositoryForAggregate } from '../../domain/repositories/interfaces/repository-for-aggregate.interface';
+import { ISpecification } from '../../domain/repositories/interfaces/specification.interface';
+import { AggregateCompositeIdentifier } from '../../domain/types/AggregateCompositeIdentifier';
+import { AggregateId } from '../../domain/types/AggregateId';
+import { AggregateType } from '../../domain/types/AggregateType';
+import { InternalError } from '../../lib/errors/InternalError';
+import { Maybe } from '../../lib/types/maybe';
+import { isNotFound } from '../../lib/types/not-found';
+import { DTO } from '../../types/DTO';
+import { ResultOrError } from '../../types/ResultOrError';
+import formatAggregateCompositeIdentifier from '../../view-models/presentation/formatAggregateCompositeIdentifier';
+import { ArangoEventRepository } from './arango-event-repository';
+
+type AggregateContextIdentifier =
+    | AggregateCompositeIdentifier
+    | Pick<AggregateCompositeIdentifier, 'type'>;
+
+export interface IEventRepository {
+    fetchEvents(aggregateContextIdentifier: AggregateContextIdentifier): Promise<BaseEvent[]>;
+
+    // TODO Should the event be an instance or DTO?
```

Comment:
Think about this now.

Reply:
I started down this path, but it's a bit of a battle. We need to make sure we leverage the event union factory within all repositories so that we can append event instances and not just event DTOs to the `eventHistory`. Let's make a separate story for this.

Reply:
[Here](https://www.pivotaltracker.com/story/show/186002801) it is.